### PR TITLE
add hasal task cmd pool

### PR DESCRIPTION
**Pull request by ShakoHo**
@askeing @MikeLien @ypwalter r?

### CODE REVIEW

File: ejenti/tasks/firefoxBuildTasks.py [hunk 1]
```diff
@@ -264,7 +264,10 @@ def deploy_fx_package(**kwargs):
     if fx_dl_pkg_path:
         if extract_fx_pkg(fx_dl_pkg_path):
             link_fx_pkg()
+            return True
         else:
             logging.error("cannot extract firefox package [%s]" % fx_dl_pkg_path)
+            return False
     else:
         logging.warn("please specify firefox downloaded package path after cmd")
+        return False
```
Comment (ypwalter) [inline ejenti/tasks/firefoxBuildTasks.py]: How about:
```
     if fx_dl_pkg_path:
          if extract_fx_pkg(fx_dl_pkg_path):
              link_fx_pkg()
              return True
          else:
              logging.error("cannot extract firefox package [%s]" % fx_dl_pkg_path)
     else:
          logging.warn("please specify firefox downloaded package path after cmd")
     return False
```

File: ejenti/tasks/baseTasks.py [hunk 2]
```diff
@@ -8,3 +11,15 @@ def init_task(kwargs):
 def parse_cmd_parameters(input_queue_msg):
     input_cmd_str = input_queue_msg.get("input_cmd_str", "")
     return [spilt_value for spilt_value in input_cmd_str.split(" ") if spilt_value != ""]
+
+
+def get_hasal_repo_path(task_config):
+    current_path_list = os.getcwd().split(os.sep)
+    for d_name in reversed(current_path_list):
```
Comment (ypwalter) [inline ejenti/tasks/baseTasks.py]:
```
    current_path_list = os.getcwd().split(os.sep)
    default_repo_path = os.sep.join(current_path_list[current_path_list.index("hasal"):])
    repo_path = task_config.get("repo_path", default_repo_path)
    return repo_path
```

File: ejenti/tasks/githubTasks.py [hunk 3]
```diff
@@ -1,27 +1,16 @@
...
-    current_path_list = os.getcwd().split(os.sep)
-    for d_name in reversed(current_path_list):
-        if d_name.lower() != "hasal":
-            current_path_list.pop()
-        else:
-            break
-    default_repo_path = os.sep.join(current_path_list)
-    repo_path = task_config.get("repo_path", default_repo_path)
-    return repo_path
+# "It was written in a time where destructors (as implemented in the __del__ method) still ran deterministically."
```
Comment (ypwalter) [inline ejenti/tasks/githubTasks.py]: L12 & L13 are dup?

File: ejenti/tasks/baseTasks.py (hunk 2 above)
Comment (ShakoHo) [inline ejenti/tasks/baseTasks.py]: it seems not working on my mac, would you please check the example code again?

Comment (ypwalter) [inline ejenti/tasks/baseTasks.py]: How about:
```python
def get_hasal_repo_path(task_config):
    current_path_list = os.getcwd().split(os.sep)
    default_repo_path = os.sep.join(current_path_list[:current_path_list.index("hasal")])
    default_repo_path.reverse()
    repo_path = task_config.get("repo_path", default_repo_path)
    return repo_path
```

**Pull request merged**